File: clients/csharp-client/Bot/Bot.cs
// Copyright (c) 2005-2016, Coveo Solutions Inc.

using System;
using System.Collections.Generic;
using System.Linq;

namespace CoveoBlitz.Bot
{
    /// <summary>
    /// Bot
    ///
    /// This bot controls your player.
    /// </summary>
    public class Bot : ISimpleBot
    {
        private readonly Random random = new Random();

        /// <summary>
        /// This will be run before the game starts
        /// </summary>
        public void Setup()
        {
            Console.WriteLine("Coveo's C# RandomBot");
        }

        /// <summary>
        /// This will be run on each turns. It must return a direction fot the bot to follow
        /// </summary>
        /// <param name="state">The game state</param>
        /// <returns></returns>
        public string Move(GameState state)
        {
            var pathfinder = new Pathfinder (state.board);

            // TODO implement SkyNet here
            // Pathfinding example:
            // string direction = pathfinder.NavigateTowards(state.myHero.pos, new Pos(0, 0));
            string direction;

            switch (random.Next(0, 5)) {
                case 0:
                    direction = Direction.East;
                    break;

                case 1:
                    direction = Direction.West;
                    break;

                case 2:
                    direction = Direction.North;
                    break;

                case 3:
                    direction = Direction.South;
                    break;

                default:
                    direction = Direction.Stay;
                    break;
            }

            Console.WriteLine("Completed turn {0}, going {1}", state.currentTurn, direction);
            return direction;
        }

        /// <summary>
        /// This is run after the game.
        /// </summary>
        public void Shutdown()
        {
            Console.WriteLine("Done");
        }
    }
}

File: clients/nodejs-client/bot.js
"use strict";

const open = require("open");

const MapUtils = require('./map');
const Types = require('./types');
const Pathfinding = require('./pathfinding');

const directions = ["north", "south", "east", "west", "stay"];
let first = true;
function bot(state, callback) {
  if (first) {
    console.log('Open Browser at ' + state.viewUrl);
    open(state.viewUrl);
    first = false;
  }

  const map = MapUtils.parseBoard(state.game.board);
  const hero = map[state.hero.pos.x][state.hero.pos.y];

  // TODO implement SkyNet here
  // Pathinding example:
  // const dir = Pathfinding.navigateTowards(map, hero, map[0][0]);
  const dir = directions[Math.floor(Math.random() * directions.length)];

  console.log(dir);
  callback(null, dir);
};


module.exports = bot;
if (require.main === module)
  require('./client/index').cli(bot);
